File: 36/api/news.js
export function getNewsByChannel(channel, start = 0, end = 10) {
    return new Promise((resolve, reject) => {
        const api = `https://is.snssdk.com/api/news/feed/v51/?category=${channel}&page=${start}`;
        fetch(api)
            .then(resp => resp.json())
            .then(json => {
                if (json.message === 'success') {
                    const newArr = json.data;
                    let list = [];
                    for (let item of newArr) {
                        list.push(JSON.parse(item.content));
                    }
                    resolve(list);
                } else {
                    throw new Error(json.status);
                }
            })
            .catch(e => {
                reject(e.toString());
            });
    });
}

File: 40/lib/api/news.dart
import 'dart:convert';
import 'dart:math';
import 'package:dio/dio.dart';
import '../model/news_response.dart';

Dio dio = Dio();

Future<List<NewsDataContent>> fetchNews(String channel) async {
  double startDouble = Random().nextDouble() * 10;
  int pageStart = startDouble.floor() * 10;
  Response res = await dio.get("http://is.snssdk.com/api/news/feed/v51/?category=${channel}&start=${pageStart}&end=${pageStart + 10}");
  NewsResponse data = NewsResponse.fromJson(res.data);
  return data.data.map((item) {
    return NewsDataContent.fromJson(jsonDecode(item.content));
  }).toList();
}
